gcsfs/core.py

```diff
@@ -1601,16 +1601,21 @@ def sign(self, path, expiration=100, **kwargs):
         URL : str
             The signed URL
         """
+        from google.api_core.client_options import ClientOptions
         from google.cloud import storage
 
-        bucket, key, generation = self.split_path(path)
         client = storage.Client(
-            credentials=self.credentials.credentials, project=self.project
+            credentials=self.credentials.credentials,
+            project=self.project,
+            client_options=ClientOptions(api_endpoint=self._endpoint),
```

Comment:
Should this include `self.session_kwargs`?

Reply:
Please check the indenting here, it looks wrong in the github diff view.

Reply:
removed it for now (for some reason on CI if was not working, it was not picking up the ClientOptions). Instead I pass api_endpoint directly to the `blob.generate_signed_url` for now. There is a way to pass some headers via `kwargs` to it, not sure if this is enough to use proxy, etc.

Reply:
Hm, I don't know either - let's leave it unless someone needs it in the future.